### PR TITLE
Support mysql engine

### PR DESCRIPTION
**Pull request by andrewkatz**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -13,8 +13,8 @@ resource "aws_security_group" "database" {
   vpc_id = var.vpc_id
 
   ingress {
-    from_port       = 5432
-    to_port         = 5432
+    from_port       = var.engine == "mysql" ? 3306 : 5432
```
Comment (trobrock) [inline main.tf]: I'd rather make the port explicitly a variable with a default that matches the default engine. That will support the possibility of custom ports. It _might_ even be possible to default one based on the other?

File: outputs.tf [hunk 2]
```diff
@@ -1,3 +1,3 @@
 output "url" {
-  value = "${aws_db_instance.database.engine}://${var.username}:${var.password}@${aws_db_instance.database.endpoint}/${var.name}"
+  value = "${var.engine == "mysql" ? "mysql2" : aws_db_instance.database.engine}://${var.username}:${var.password}@${aws_db_instance.database.endpoint}/${var.name}"
```
Comment (trobrock) [inline outputs.tf:2]: Does just mysql not work here? I thought it did.

**Review by trobrock — CHANGES_REQUESTED** [review]

File: outputs.tf (hunk 2 above)
Comment (andrewkatz) [reply outputs.tf:2]: It doesn't for Rails when using the `DATABASE_URL` environment variable. It tries to use the `mysql` adapter, which doesn't exist.

Alternatively, we could change this to be `mysql`, keeping it framework agnostic, then update the environment variable for the app by doing a `replace(module.database.url, "mysql://", "mysql2://")`.

File: main.tf (hunk 1 above)
Comment (andrewkatz) [reply main.tf]: I do like that better. I originally tried doing that with:

```terraform
variable "port" {
  default = var.engine == "mysql" ? 3306 : 5432
}
```

but I got an error saying I can't use variables there. But we could make it default to `-1` or something, falling back to engine-specific defaults in `main.tf`.

Comment (andrewkatz) [reply main.tf]: Updated to allow custom ports, falling back to engine defaults.

**Review by trobrock — APPROVED** [review]

**Pull request merged**